shop3/looping-tasks/exts-common.lisp

```diff
@@ -0,0 +1,89 @@
+(in-package :shop)
+
+(defclass looping-domain (domain)
+     ()
+  (:documentation "Mixin domain class for HTNs with LOOP tasks in them."))
+
+
+(defmethod seek-plans-task ((domain looping-domain) task1 state tasks
+                            top-tasks
+                            partial-plan
+                            partial-plan-cost depth which-plans
+                            protections
+                            unifier)
+  (trace-print :tasks (get-task-name task1) state
+               "~2%Depth ~s, trying task ~s"
+               depth
+               (apply-substitution task1 unifier))
+  ;;    (y-or-n-p "continue?")
+  (cond
+    ((primitivep (get-task-name task1))
+     (seek-plans-primitive domain task1 state tasks top-tasks
+                           partial-plan partial-plan-cost depth which-plans
+                           protections
+                           unifier))
+
+    ;; I can directly design this in the NONPRIMITIVE but just not to
+    ;; complicate that function and possibly break it.
+    ((looping-p (get-task-name task1))
+     (seek-plans-loop domain task1 state tasks top-tasks
+                      partial-plan partial-plan-cost depth
+                      which-plans protections
+                      unifier))
+    
+    (t
+     (seek-plans-nonprimitive domain task1 state tasks top-tasks
+                                     partial-plan partial-plan-cost depth
+                                     which-plans protections
+                                     unifier
+                                     ))))
+
+(defun save-reduction (ess-search-state reduction)
+  (when *enhanced-plan-tree*
+    (with-slots (backtrack-stack plan-tree-lookup) ess-search-state
+      (iter
+       (for task in reduction)
+       (let* ((parent (plan-tree:find-task-in-tree (current-task ess-search-state)
+						   (plan-tree-lookup ess-search-state)))
+	      (child (make-plan-tree-for-task-net task
+						  parent plan-tree-lookup)))
+	 
+	 ;; MAKE-PLAN-TREE-FOR-TASK-NET as a side-effect, links PARENT and CHILD.
+	 (push (make-add-child-to-tree :parent parent
+				       :child child)
+	       backtrack-stack)))))
+  ess-search-state)
+
+(defun new-symbol (sym)
```

Comment:
**Never** use `intern` without specifying the optional `package` argument, otherwise you just get whatever `*package*` is bound to when `new-symbol` is called.

Also `gentemp` = `gensym` followed by `intern`, so just use `gentemp`